src/lib.rs

```diff
@@ -586,22 +593,54 @@ mod tests {
     fn parse_csi_max_params() {
         use crate::MAX_PARAMS;
 
-        static INPUT: &[u8] = b"\x1b[1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;p";
+        // This will build a list of repeating '1;'s
+        // The length is MAX_PARAMS - 1 because the last semicolon is interpreted
+        // as an implicit zero, making the total number of parameters MAX_PARAMS
+        let params = std::iter::repeat("1;").take(MAX_PARAMS - 1).collect::<std::string::String>();
+        let input = format!("\x1b[{}p", &params[..]).into_bytes();
 
         // Create dispatcher and check state
         let mut dispatcher = CsiDispatcher::default();
         assert!(!dispatcher.dispatched_csi);
 
-        // Run parser using OSC_BYTES
+        // Run parser using INPUT
         let mut parser = Parser::new();
-        for byte in INPUT {
-            parser.advance(&mut dispatcher, *byte);
+        for byte in input {
+            parser.advance(&mut dispatcher, byte);
+        }
+
+        // Check that flag is set and thus csi_dispatch assertions ran.
+        assert!(dispatcher.dispatched_csi);
+        assert_eq!(dispatcher.params.len(), 1);
+        assert_eq!(dispatcher.params[0].len(), MAX_PARAMS);
+        assert!(!dispatcher.ignore);
+    }
+
+    #[test]
+    fn parse_csi_params_ignore_long_params() {
+        use crate::MAX_PARAMS;
```

Comment:
We have this in multiple tests now, but I don't see why we wouldn't add it at the top of the `test` module. Using just `use super::*;` would be fine too, since that's a pretty common pattern for test modules (at least in the repos of the Alacritty project).